{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Assigning TREs with their putative target genes using Hi-C"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Downloading public Hi-C datasets\n",
    "\n",
    "I downloaded the raw fastq files from Hi-C datasets that were obtained from several non-treated primate LCLs.\n",
    "\n",
    "### Human LCL\n",
    "\n",
    "I obtained the Hi-C datasets from human LCL ID (GM12878) from two different papers:\n",
    "\n",
    "1) https://doi.org/10.1016/j.cell.2015.07.048\n",
    "\n",
    "```\n",
    "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE62742\n",
    "    SRR1802420\n",
    "    SRR1802421\n",
    "    SRR1802422\n",
    "    SRR1802423\n",
    "    SRR1802424\n",
    "    SRR1802425\n",
    "    SRR1802426\n",
    "Total ~1,126 million paired-end reads\n",
    "\n",
    "cat SRR1802420_1.fastq SRR1802421_1.fastq SRR1802422_1.fastq SRR1802423_1.fastq \\\n",
    "SRR1802424_1.fastq SRR1802425_1.fastq SRR1802426_1.fastq > HiC-Human_1.fastq\n",
    "cat SRR1802420_2.fastq SRR1802421_2.fastq SRR1802422_2.fastq SRR1802423_2.fastq \\\n",
    "SRR1802424_2.fastq SRR1802425_2.fastq SRR1802426_2.fastq > HiC-Human_2.fastq\n",
    "```\n",
    "2) https://doi.org/10.1016/j.cell.2014.11.021\n",
    "```\n",
    "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE63525\n",
    "\n",
    "SRAs for dilution Hi-C protocol\n",
    "    SRR1658643\n",
    "    SRR1658617\n",
    "    SRR1658620\n",
    "    SRR1658622\n",
    "    SRR1658608\n",
    "    SRR1658621\n",
    "    SRR1658609\n",
    "    SRR1658618\n",
    "    SRR1658610\n",
    "    SRR1658619\n",
    "Total ~2,740 billion paired-end reads\n",
    "\n",
    "cat SRR1658643_1.fastq SRR1658617_1.fastq SRR1658620_1.fastq SRR1658622_1.fastq \\\n",
    "SRR1658608_1.fastq SRR1658621_1.fastq SRR1658609_1.fastq SRR1658618_1.fastq SRR1658610_1.fastq \\\n",
    "SRR1658619_1.fastq > HiC-Human-dilution_1.fastq\n",
    "cat SRR1658643_2.fastq SRR1658617_2.fastq SRR1658620_2.fastq SRR1658622_2.fastq \\\n",
    "SRR1658608_2.fastq SRR1658621_2.fastq SRR1658609_2.fastq SRR1658618_2.fastq SRR1658610_2.fastq \\\n",
    "SRR1658619_2.fastq > HiC-Human-dilution_2.fastq\n",
    "\n",
    "SRAs for in-situ Hi-C protocol\n",
    "    SRR1658572\n",
    "    SRR1658592\n",
    "    SRR1658600\n",
    "    SRR1658599\n",
    "    SRR1658601\n",
    "    SRR1658579\n",
    "    SRR1658583\n",
    "    SRR1658588\n",
    "    SRR1658596\n",
    "    SRR1658578\n",
    "Total ~2,900 billion paired-end reads\n",
    "\n",
    "cat SRR1658572_1.fastq SRR1658592_1.fastq SRR1658600_1.fastq SRR1658599_1.fastq \\\n",
    "SRR1658601_1.fastq SRR1658579_1.fastq SRR1658583_1.fastq SRR1658588_1.fastq SRR1658596_1.fastq \\\n",
    "SRR1658578_1.fastq > HiC-Human-insitu_1.fastq\n",
    "cat SRR1658572_2.fastq SRR1658592_2.fastq SRR1658600_2.fastq SRR1658599_2.fastq \\\n",
    "SRR1658601_2.fastq SRR1658579_2.fastq SRR1658583_2.fastq SRR1658588_2.fastq SRR1658596_2.fastq \\\n",
    "SRR1658578_2.fastq > HiC-Human-insitu_2.fastq\n",
    "```\n",
    "### Chimp LCL\n",
    "\n",
    "I obtained the Hi-C datasets from chimp LCL from this paper:<br>\n",
    "https://doi.org/10.1016/j.cels.2019.05.011\n",
    "```\n",
    "https://www.ncbi.nlm.nih.gov/sra?term=SRX5572056\n",
    "    SRR8782019\n",
    "    SRR8782020\n",
    "    SRR8782021\n",
    "Total ~530 million paired-end reads\n",
    "\n",
    "cat SRR8782019_1.fastq SRR8782020_1.fastq SRR8782021_1.fastq > HiC-Chimp_1.fastq\n",
    "cat SRR8782019_2.fastq SRR8782020_2.fastq SRR8782021_2.fastq > HiC-Chimp_2.fastq\n",
    "```\n",
    "### Bonobo LCL\n",
    "\n",
    "I obtained the Hi-C datasets from Bonobo LCL (Kidogo) from this paper:<br>\n",
    "https://doi.org/10.1016/j.cels.2019.05.011\n",
    "```\n",
    "https://www.ncbi.nlm.nih.gov/sra?term=SRX5572057\n",
    "    SRR8782022\n",
    "    SRR8782023\n",
    "    SRR8782024\n",
    "Total ~515 million paired-end reads\n",
    "\n",
    "cat SRR8782022_1.fastq SRR8782023_1.fastq SRR8782024_1.fastq > HiC-Bonobo_1.fastq\n",
    "cat SRR8782022_2.fastq SRR8782023_2.fastq SRR8782024_2.fastq > HiC-Bonobo_2.fastq\n",
    "```\n",
    "### Gorilla LCL\n",
    "\n",
    "I obtained the Hi-C datasets from Gorilla LCL from this paper:<br>\n",
    "https://doi.org/10.1016/j.cels.2019.05.011\n",
    "```\n",
    "https://www.ncbi.nlm.nih.gov/sra?term=SRX5572058\n",
    "    SRR8782025\n",
    "    SRR8782026\n",
    "    SRR8782027\n",
    "Total ~423 million paired-end reads\n",
    "\n",
    "cat SRR8782025_1.fastq SRR8782026_1.fastq SRR8782027_1.fastq > HiC-Gorilla_1.fastq\n",
    "cat SRR8782025_2.fastq SRR8782026_2.fastq SRR8782027_2.fastq > HiC-Gorilla_2.fastq\n",
    "```\n",
    "### Gibbon LCL\n",
    "\n",
    "I obtained the Hi-C datasets from Gibbon LCL (Vok) from this paper:<br>\n",
    "http://www.genome.org/cgi/doi/10.1101/gr.233874.117\n",
    "```\n",
    "https://www.ncbi.nlm.nih.gov/sra?term=SRP149294\n",
    "    SRR7232221\n",
    "Total ~385 million paired-end reads\n",
    "\n",
    "mv SRR7232221_1.fastq HiC-Gibbon_1.fastq\n",
    "mv SRR7232221_2.fastq HiC-Gibbon_2.fastq\n",
    "```\n",
    "### Rhesus LCL\n",
    "\n",
    "I obtained the Hi-C datasets from Rhesus LCL (Mm290-96) from this paper:<br>\n",
    "http://www.genome.org/cgi/doi/10.1101/gr.265322.120\n",
    "```\n",
    "https://www.ebi.ac.uk/ena/browser/view/PRJEB37908\n",
    "    ERR4501103\n",
    "    ERR4501104\n",
    "Total ~524 million paired-end reads\n",
    "\n",
    "cat ERR4501103_1.fastq ERR4501104_1.fastq > HiC-Rhesus_1.fastq\n",
    "cat ERR4501103_2.fastq ERR4501104_2.fastq > HiC-Rhesus_2.fastq\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Hi-C processing and analysis software\n",
    "\n",
    "- HiC-PRO (https://github.com/nservant/HiC-Pro)\n",
    "- Fit-Hi-C (https://github.com/ay-lab/fithic)\n",
    "- cLoops (https://github.com/YaqiangCao/cLoops)\n",
    "- Peakachu (https://github.com/tariks/peakachu)\n",
    "- HiC-bench (https://github.com/NYU-BFX/hic-bench)\n",
    "\n",
    "Installing/testing:\n",
    "- Juicer/Hiccups (https://github.com/aidenlab/juicer)\n",
    "\n",
    "https://github.com/aidenlab/juicer/wiki/Installation <br>\n",
    "https://github.com/aidenlab/juicer/wiki/Running-Juicer-on-a-cluster"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## LiftOver between b37 and hg38\n",
    "\n",
    "https://gatk.broadinstitute.org/hc/en-us/articles/360035890951 <br>\n",
    "https://gatk.broadinstitute.org/hc/en-us/articles/360035890711?id=23390 <br>\n",
    "https://raw.githubusercontent.com/broadinstitute/gatk/master/scripts/funcotator/data_sources/gnomAD/b37ToHg38.over.chain"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Using already processed Hi-C loops.\n",
    "\n",
    "```\n",
    "wget https://ftp.ncbi.nlm.nih.gov/geo/series/GSE63nnn/GSE63525/suppl/GSE63525_GM12878_combined_intrachromosomal_contact_matrices.tar.gz\n",
    "wget https://ftp.ncbi.nlm.nih.gov/geo/series/GSE63nnn/GSE63525/suppl/GSE63525_GM12878_primary+replicate_HiCCUPS_looplist.txt.gz\n",
    "wget https://ftp.ncbi.nlm.nih.gov/geo/series/GSE63nnn/GSE63525/suppl/GSE63525_GM12878_primary_README.rtf\n",
    "wget https://ftp.ncbi.nlm.nih.gov/geo/series/GSE63nnn/GSE63525/suppl/GSE63525_CH12_combined_30.hic\n",
    "\n",
    "# Editing HiCCUPS_looplist.txt onto a longrange file for visualization.\n",
    "# Removes the first header line.\n",
    "tail -n +2 GSE63525_GM12878_primary+replicate_HiCCUPS_looplist.txt > test0.bed\n",
    "# Keeps only the columns with the coordinates, and the score.\n",
    "awk -v OFS='\\t' '{print \"chr\"$1, $2, $3, \"chr\"$4, $5, $6, $8}' test0.bed > test1.bed\n",
    "# Keeps only entries with a score greater than 100.\n",
    "awk -v OFS='\\t' '{if ($7 > 100) print $0}' test1.bed > test2.bed\n",
    "# Edits entries so that they resemble the 'longrange' format.\n",
    "awk -v OFS='\\t' '{ print $1, $2, $3, $4\":\"$5\"-\"$6\",\"$7\"\\n\"$4, $5, $6, $1\":\"$2\"-\"$3\",\"$7}' test2.bed > test3.bed\n",
    "# Sorts entries.\n",
    "sort -k1,1 -k2,2n test3.bed > test4.bed\n",
    "#Compresses with bgzip.\n",
    "module load samtools/1.10\n",
    "bgzip test4.bed\n",
    "#Creates index with samtools tabix.\n",
    "tabix -p bed test4.bed.gz\n",
    "```"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
